test/src/org/labkey/test/components/mobileappstudy/TokenBatchesWebPart.java

```diff
@@ -49,7 +49,10 @@ public boolean isNewBatchPresent()
 
     public boolean isNewBatchEnabled()
     {
-        return elementCache().newBatchButton.isEnabled();
+        return !elementCache().newBatchButton.getAttribute("class").contains("labkey-disabled-button");
+
+        // NOTE: This does not work... wonder why
+        // return elementCache().newBatchButton.isEnabled();
```

Comment:
To answer your question. `WebElement.isEnabled` just looks for [the `disabled` attribute](https://html.spec.whatwg.org/#concept-fe-disabled) on the element. It doesn't know all of JavaScript-y ways one might make a button not do anything.

Reply:
Okay, so I guess I'll remove the initial implementation... looks like it wasn't used and probably never worked.

Reply:
Removed comment